### PR TITLE
Nushell table list columns -> dataframe list columns. Explode / Flatten dataframe support.

### PR DESCRIPTION
**Pull request by ayax79**
# Description
- Adds support for conversion between nushell lists and polars lists instead of treating them as a polars object. 
- Fixed explode and flatten to work both as expressions or lazy dataframe commands. The previous item was required to make this work.

### CODE REVIEW

**Comment by amtoine** [comment]
@ayax79 
marking this as a draft until it's ready :wink:

**Comment by fdncred** [comment]
nice start @ayax79! looking forward to seeing more.

**Comment by fdncred** [comment]
still putting in the hours I see. keep up the good work!

File: crates/nu-cmd-dataframe/src/dataframe/lazy/explode.rs [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+                            Value::test_string("Knitting"),
+                            Value::test_string("Skiing"),
+                            Value::test_string("Football"),
+                        ]),
+                   ]).expect("simple df for test should not fail")
+                   .into_value(Span::test_data()),
+                    )
+            },
+            Example {
+                description: "todo expression case",
```
Comment (fdncred) [inline crates/nu-cmd-dataframe/src/dataframe/lazy/explode.rs:77]: Can we have a better name here other than `todo`?

File: crates/nu-cmd-dataframe/src/dataframe/values/nu_dataframe/conversion.rs [hunk 2]
```diff
@@ -3,16 +3,27 @@ use super::{DataFrameValue, NuDataFrame};
...
 };
 use std::ops::{Deref, DerefMut};
 
 const SECS_PER_DAY: i64 = 86_400;
 
+// The values capacity is for the size of an internal vec.
+// Since this is impossible to determine without traversing every value
+// I just picked one. Since this is for converting back and forth
+// between nushell tables the values shouldn't be too extremely large for
+// practical reasone (~ a few thousand rows).
```
Comment (fdncred) [inline crates/nu-cmd-dataframe/src/dataframe/values/nu_dataframe/conversion.rs:24]:
```suggestion
// practical reasons (~ a few thousand rows).
```

File: crates/nu-cmd-dataframe/src/dataframe/lazy/flatten.rs [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+                            Value::test_string("Knitting"),
+                            Value::test_string("Skiing"),
+                            Value::test_string("Football"),
+                        ]),
+                   ]).expect("simple df for test should not fail")
+                   .into_value(Span::test_data()),
+                    )
+            },
+        Example {
+                description: "todo expression case",
```
Comment (fdncred) [inline crates/nu-cmd-dataframe/src/dataframe/lazy/flatten.rs:79]: Maybe a better name here other than `todo` too. :)

File: crates/nu-cmd-dataframe/src/dataframe/lazy/flatten.rs [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+                    Type::Custom("dataframe".into()),
+                ),
+            ])
+            .category(Category::Custom("lazyframe".into()))
+    }
+
+    fn examples(&self) -> Vec<Example> {
+        vec![
+Example {
+                description: "Explode the specified dataframe",
```
Comment (fdncred) [inline crates/nu-cmd-dataframe/src/dataframe/lazy/flatten.rs:46]:
```suggestion
                description: "Flatten the specified dataframe",
```

**Comment by fdncred** [comment]
I added a few comments, but I'm wondering if identical tests for `flatten` and `explode` is appropriate. It's hard to tell if things are working or not because they're the same.

File: crates/nu-cmd-dataframe/src/dataframe/lazy/explode.rs (hunk 1 above)
Comment (ayax79) [reply crates/nu-cmd-dataframe/src/dataframe/lazy/explode.rs:77]: I forgot to remove after I added the Example.. fixing.

**Comment by ayax79** [comment]
> I added a few comments, but I'm wondering if identical tests for `flatten` and `explode` is appropriate. It's hard to tell if things are working or not because they're the same.

I could tweak them a bit. The polars docs say that flatten is just an alias for explode.

**Comment by fdncred** [comment]
> The polars docs say that flatten is just an alias for explode.

Oh, I didn't know that. Thanks!

**Comment by fdncred** [comment]
Let's move forward with this. We appreciate your help!

**Pull request merged**